### ISSUE TITLE
error: timeout on transient error: Could not connect to the server 127.0.0.1:8333 (error code 1 - "EOF reached")

### ISSUE THREAD

**Issue by GaloisField2718** [bitcoin/bitcoin#27121]
When I'm trying to use `bitcoin-cli` while bitcoind running it displays error message : 
```error: timeout on transient error: Could not connect to the server 127.0.0.1:8333 (error code 1 - "EOF reached")
Make sure the bitcoind server is running and that you are connecting to the correct RPC port.```

<!--- What behavior did you expect? -->

running command : `bitcoin-cli getblockchaininfo` 

<!--- What was the actual behavior (provide screenshots if the issue is GUI-related)? -->

` bitcoind -txindex `
```[...] 2023-02-17T18:03:20Z Bound to 127.0.0.1:8334
2023-02-17T18:03:20Z Bound to [::]:8333
2023-02-17T18:03:20Z Bound to 0.0.0.0:8333
[...] ```

<!--- How reliably can you reproduce the issue, what are the steps to do so? -->

I am on MacOS Ventura 13.1 intel i5 MacBook Pro.

<!-- What version of Bitcoin Core are you using, where did you get it (website, self-compiled, etc)? -->

`2023-02-17T18:03:07Z Bitcoin Core version v24.0.1 (release build)`


<!-- Any extra information that might be useful in the debugging process. -->

I saw a related issue few years ago and apparently this issue should to be patched.

**Comment by GaloisField2718** [comment]
Hello, 

Now it's solved but initially it was only : 
launch `bitcoind`
In another window launching `bitcoin-cli getblockchaininfo` command. 

What weird was the absence of message : `bitcoin-cli` command not found.

Because now I think it's because I didn't call the path where is bin/bitcoin-cli.

Or maybe because the chain didn't finished to load and so I should wait.

I don't know exactly but now it's solve I will close it. 

Thank you very much for your answer !!🙌

**Comment by epiccurious** [comment]
Adding more context to this closed issue in case other people stumble on it too.

I was able to reproduce this error with a script running Bitcoin Core 25.0 and 24.0.1 (x86_64-linux-gnu on Ubuntu).

Steps to reproduce:
1. Delete the data directory.
2. Launch `bitcoin-qt --server`.
3. Immediately check the chain state with `bitcoin-qt getblockchaininfo`.

The RPC server doesn't have enough time to initialize, so the call to port 8333 fails.

Fixed the issue by using the rpcwait option:
`bitcoin-cli --rpcwait getblockchaininfo`
